server/src/main/java/org/opensearch/action/admin/indices/segments/IndicesSegmentResponse.java

```diff
@@ -87,20 +85,19 @@ public Map<String, IndexSegments> getIndices() {
         }
         Map<String, IndexSegments> indicesSegments = new HashMap<>();
 
-        Set<String> indices = new HashSet<>();
+        Map<String, List<ShardSegments>> indicesShardSegments = new HashMap<>();
         for (ShardSegments shard : shards) {
```

Comment:
If we sort the ShardSegments by `indexName` - can we avoid the double iteration ?